client/src/elements/schema-editor/schema-editor-table.js

```diff
@@ -0,0 +1,144 @@
+import { bindable, inject, Loader } from 'aurelia-framework';
+import array2d from 'array2d';
+
+function hasNonNullInArray(arr) {
+  for (let val of arr) {
+    if (val !== null) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function trimNull(data) {
+  let lastColumnWithValues = 0;
+  array2d.eachColumn(data, (column, index) => {
+    if (hasNonNullInArray(column)) {
+      lastColumnWithValues = index;
+    }
+  });
+
+  let lastRowWithValues = 0;
+  array2d.eachRow(data, (row, index) => {
+    if (hasNonNullInArray(row)) {
+      lastRowWithValues = index;
+    }
+  });
+
+  return array2d.crop(data, 0, 0, lastColumnWithValues + 1, lastRowWithValues + 1);
+}
+
+function emptyToNull(data) {
+  array2d.eachCell(data, (cell, i, j) => {
+    if (cell === '' || cell === undefined) {
+      data[i][j] = null;
+    }
+  });
+  return data;
+}
+
+@inject(Loader)
+export class SchemaEditorTable {
+
+  @bindable schema;
+  @bindable data;
+  @bindable change;
+
+  constructor(loader) {
+    this.loader = loader;
+  }
+
+  dataChanged() {
+    this.hotData = JSON.parse(JSON.stringify(this.data));
+  }
+
+  async attached() {
+    this.loader.loadModule('handsontable/dist/handsontable.full.css!');
+    const Handsontable = await this.loader.loadModule('handsontable/dist/handsontable.full.js');
+
+    this.hot = new Handsontable(this.tableContainerElement, {
+      dataSchema: [],
+      height: this.getGridHeight(),
+      minRows: 5,
+      minCols: 5,
+      minSpareRows: 1,
+      minSpareCols: 1,
+      rowHeaders: true,
+      colHeaders: false,
+      contextMenu: false,
+      colWidths: 70,
+      rowHeights: 23,
+      afterChange: (changes, source) => {
+        if (this.hot) {
+          this.replaceCommaWithPointIfDecimal();
+          this.hot.updateSettings({
+            height: this.getGridHeight()
+          });
+        }
+        if (source !== 'loadData') {
+          this.data = trimNull(emptyToNull(this.hot.getData()));
+          this.change();
+        }
+      }
+    });
+    if (Array.isArray(this.hotData)) {
+      this.hot.loadData(this.hotData);
+    }
+  }
+
+  getGridHeight() {
+    return Math.min(280, Math.max(5, this.data.length) * 23) + 30;
+  }
+
+  replaceCommaWithPointIfDecimal() {
```

Comment:
@fuenkchen thanks for spotting the bug. this is fixed with 6312bcc48968d69f5c688b8a6644f0c4a883801a